deps/fs.lua

```diff
@@ -577,3 +577,31 @@ end
 function fs.createReadStream(path, options)
   return fs.ReadStream:new(path, options)
 end
+function fs.appendFile(filename, data, callback)
+  callback = callback or function() end
+  local function write(fd, offset, buffer, callback)
+    local function onWrite(err, written)
+      if err then return fs.close(fd, function() callback(err) end) end
+      if written == #buffer then
+        fs.close(fd, callback)
+      else
+        if offset == -1 then offset = 0 end
+        offset = offset + written
+        write(fd, offset, buffer, callback)
```

Comment:
Do we still want to write the full buffer in this case or do we want a substring of it?